### ISSUE TITLE
[grouping]: Config transition won't run for projects relying on default config option

### ISSUE THREAD

**Issue by lobsterkatie** [getsentry/sentry#78282]
TL;DR: Projects which don't have a grouping config projectoption set will miss out on the config transition feature because we won't have any way to know they haven't been on the default all along. We therefore need to store a grouping config on every project, even if it matches the default.

--------------

Right now our config transition logic works like this:

- When an event comes in, check if the project is on an outdated config and upgrade it if so, setting the outdated config as the project's secondary config.
- When calculating hashes, only calculate a secondary hash if the project has a secondary config set (and its transition expiry date hasn't passed).

The problem comes in that "check if the project is on an outdated config" part. If the project doesn't have a grouping config option set, and just relies on defaulting to the current config, it will never read as outdated and therefore never be kicked into a transition period, thereby losing out on the group continuity the transition feature is meant to achieve.

Of the ~2.3 million projects in our US database, only ~118K have a grouping config set, so this problem - while currently dormant - is quite widespread. Whenever we next move to a new config, it will end up affecting roughly 95% of our projects.

To fix this, we need to:

- On project creation, set the project option to the current config.
- Run a data migration to backfill the option for existing projects.
